core/src/main/java/io/opentelemetry/android/export/BufferDelegatingLogExporter.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.export
+
+import io.opentelemetry.sdk.common.CompletableResultCode
+import io.opentelemetry.sdk.logs.data.LogRecordData
+import io.opentelemetry.sdk.logs.export.LogRecordExporter
+
+/**
+ * An in-memory buffer delegating log exporter that buffers log records in memory until a delegate is set.
+ * Once a delegate is set, the buffered log records are exported to the delegate.
+ *
+ * The buffer size is set to 5,000 log entries by default. If the buffer is full, the exporter will drop new log records.
+ */
+internal class BufferDelegatingLogExporter(
+    maxBufferedLogs: Int = 5_000,
+) : BufferedDelegatingExporter<LogRecordData, LogRecordExporter>(bufferedSignals = maxBufferedLogs),
```

Comment:
I've experience so much pain from unnecessarily complicated class inheritance problems over my career, so it will often make me sad to see inheritance used primarily as a means to code reuse. Delegation is hard here without a common interface for log/span export, I get it, but maybe we can collectively circle back on this after this PR work is complete.

Reply:
Yeah agreed. In this case the main benefits is that you don't have to implement some methods but the delegation pattern would probably work with little overhead if you prefer we use that instead.